src/service.rs

```diff
@@ -59,22 +44,72 @@ pub struct Options {
 //
 #[allow(clippy::type_complexity)]
 #[derive(Debug)]
-pub struct GlobalState<Types: BuilderType> {
+pub struct GlobalState<Types: NodeType> {
+    // identity keys for the builder
+    // May be ideal place as GlobalState interacts with hotshot apis
+    // and then can sign on responsers as desired
+    pub builder_keys: (
+        Types::SignatureKey,                                             // pub key
+        <<Types as NodeType>::SignatureKey as SignatureKey>::PrivateKey, // private key
+    ),
+
+    // data store for the blocks
     pub block_hash_to_block: HashMap<
         BuilderCommitment,
         (
             Types::BlockPayload,
-            <<Types as BuilderType>::BlockPayload as BlockPayload>::Metadata,
-            Arc<JoinHandle<()>>,
-            <<Types as BuilderType>::SignatureKey as SignatureKey>::PureAssembledSignatureType,
-            Types::SignatureKey,
+            <<Types as NodeType>::BlockPayload as BlockPayload>::Metadata,
+            //Arc<JoinHandle<()>>,
+            Arc<JoinHandle<<VidScheme as VidSchemeTrait>::Commit>>,
+            //JoinHandle<<VidScheme as VidSchemeTrait>::Commit>,
         ),
     >,
+    // sending a request from the hotshot to the builder states
     pub request_sender: BroadcastSender<MessageType<Types>>,
-    pub response_receiver: UnboundedReceiver<ResponseMessage<Types>>,
+
+    // getting a response from the builder states based on the request sent by the hotshot
+    pub response_receiver: UnboundedReceiver<ResponseMessage>,
+
+    // sending a transaction from the hotshot/private mempool to the builder states
+    // NOTE: Currently, we don't differentiate between the transactions from the hotshot and the private mempool
+    pub tx_sender: BroadcastSender<MessageType<Types>>,
+
+    // sending a DA proposal from the hotshot to the builder states
+    pub da_sender: BroadcastSender<MessageType<Types>>,
```

Comment:
Why do we have these, other than the tx_sender?

Reply:
Since `global state` implements `hotshot events stream service` aka run_standalone_builder_service now, or should we take out `hotshot events stream service` out and keep it as separate function as earlier?

Reply:
It will also help in spawning a task based on this standalone function.

Reply:
But we need `tx_sender` in the Global State to implement `submit_txn` for txn submit api.

Reply:
Done!